test/jasmine/tests/parcoords_test.js

```diff
@@ -868,7 +871,65 @@ describe('parcoords basic use', function() {
 
     });
 
-    it('@gl Calling `Plotly.restyle` with a string path should amend the preexisting parcoords', function(done) {
+    function _getGrayRatio(secondPass, msg)
+        var canvases = d3.selectAll('.gl-canvas');
+        expect(canvases.size()).toBe(3, msg);
+        
+        canvases.each(function(element, index) {
+          
+            if (index == 0) { // FIXME: we assumed here that the context is the first item but may be not.
+            
+                var imageArray = readPixel(this, 0, 0, this.width, this.height);
+                var totalRGB = 0;
+                var n = imageArray.length;
+                for(var i = 0; i < n; i++) {
+                    var r = imageArray[i][0];
+                    var g = imageArray[i][1];
+                    var b = imageArray[i][2];
+                    totalRGB += r + g + b;
+                }
+                if(secondPass > 0) {
```

Comment:
I don't like this pattern. Let's keep `getGrayRatio` as just a _getter_ function and move the `expect` statement to down to `it` block below:

```js
        var totalRGB;
        
        Plotly.plot(gd, mock3)
        .then(function() {
            totalRGB = _getGrayRatio();

            mockCopy.data[0].dimensions[1].constraintrange = [0.4, 0.6];          
            return Plotly.react(gd, mockCopy);
        })
        .then(function() {
            var totalRGB2 = _getGrayRatio();
            expect(totalRGB2).toBe(totalRGB, 'after react');
        })
        .catch(failTest)
        .then(done);
```